packages/nova-types/src/nova-graphql.interface.ts

```diff
@@ -64,15 +63,51 @@ export interface NovaGraphQL<GraphQLDocument = any> {
     onError?: (error: Error) => void;
   }) => void;
 
-  useMutation?: (
+  useMutation?: <TMutationPayload extends OperationType>(
     mutation: GraphQLDocument,
   ) => [
     (options: {
       variables: { [name: string]: unknown };
       context?: { [name: string]: unknown };
       optimisticResponse?: unknown | null;
+      onCompleted?: ((response: TMutationPayload["response"], errors: PayloadError[] | null) => void | null) | undefined;
+      onError?: (error: Error) => void;
+    }) => Disposable,
+    boolean,
+  ];
+
+  useMutation_deprecated?: (
+    mutation: GraphQLDocument,
+  ) => [
+    (options: {
+      variables: { [name: string]: unknown };
+      context?: { [name: string]: unknown };
+      optimisticResponse?: Partial<any["response"]> | null;
       onCompleted?: (response: unknown) => void;
     }) => Promise<{ errors?: readonly Error[]; data?: unknown }>,
     boolean,
   ];
 }
+
+export type Disposable = {
+  dispose(): void;
+};
+
+export interface OperationType {
+  readonly variables: { [name: string]: any };
+  readonly context?: { [name: string]: any };
+  readonly response: any;
+  readonly rawResponse?: unknown;
+}
```

Comment:
I assume these are also this loosely typed in the Relay typings?

Reply:
In Relay they have "any" for variables, but "unknown" for the rest. I updated it to be unknown for all since it will be a bit better.